### ISSUE TITLE
Improve formatting of SQL rendered by window functions

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#1962]
The current formatting rules lead to weird SQL rendering, e.g.

``` sql
select top 100 percent [dbo].[t_book].[ID], row_number() over (order by 
  ([dbo].[t_book].[AUTHOR_ID] * 2) asc, 
  lower([dbo].[t_book].[TITLE]) asc)
from [dbo].[t_book]
```

**Comment by lukaseder** [comment]
Things are even worse when various clauses are combined in the `OVER()` clause:`

```sql
count(*) over (partition by 
  A, 
  B
 order by 
  C, 
  D
 rows unbounded preceding)
```

**Comment by lukaseder** [comment]
This will be fixed with #9764. No backport is scheduled
